tools/Microsoft.AspNetCore.Server.Kestrel.GeneratedCode/KnownHeaders.cs

```diff
@@ -31,19 +31,27 @@ static string If(bool condition, Func<string> formatter)
                     case {byLength.Key}:
                         {{{Each(byLength, header => $@"
                             if ({header.EqualIgnoreCaseBytes()})
-                            {{
+                            {{{(header.Identifier == "ContentLength" ? $@"
+                                if (ContentLength.HasValue)
+                                {{
+                                    ThrowInvalid{(className == "FrameResponseHeaders" ? "Response" : "Request")}ContentLengthException(AppendValue(HeaderUtilities.FormatInt64(ContentLength.Value), value).ToString());
```

Comment:
Throwing a BadHttpRequestException should be sufficient, since it's supposed to be caught and logged [in RequestProcessingAsync](https://github.com/aspnet/KestrelHttpServer/blob/10fe5e6fa25e95ab9acd2b3bd733e78be4315a37/src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/FrameOfT.cs#L199).

It looks like `SetBadRequestState` no longer logs the error and our tests didn't catch it though... 😢